modules/data-entry/src/main/frontend/src/questionnaire/DateQuestionUtilities.jsx

```diff
@@ -171,10 +171,36 @@ export default class DateQuestionUtilities {
     return dateString?.replace(/[-+][0-9]{2}:[0-9]{2}$/gm, '');
   }
 
+  static stripFrom(dateString, delimiter) {
+    // Remove everything after the delimiter
+    return (dateString?.includes(delimiter) ? dateString?.substring(0, dateString?.indexOf(delimiter)) : dateString)
+  }
+
+  static strip(dateAsISO, textFieldType) {
+    if (textFieldType == 'datetime-local') return this.stripFrom(dateAsISO, '.');
+    if (textFieldType == 'date') return this.stripFrom(dateAsISO, 'T');
+    return dateAsISO;
+  }
+
   static isAnswerComplete(answers, type) {
     return type == this.INTERVAL_TYPE && answers.length == 2 || answers.length == 1;
   }
 
+  static getAnswerValueInstructions(lowerLimit, upperLimit, format) {
+    if (lowerLimit || upperLimit) {
+      let min = lowerLimit?.toFormat(format);
+      let max = upperLimit?.toFormat(format);
+      if (min && max) {
+        return `Between ${min} and ${max}`;
+      } else if (min) {
+        return `${min} or later`;
+      } else {
+        return `At most ${max}`;
```

Comment:
I don't like `At most`, how about `Before ${max}`?

Reply:
"Before ${max}" suggests that `${max}` is excluded. Either we exclude `${max}` in the comparison. or we use wording that makes it clear it is included.

Reply:
Apply this if you agree:

```suggestion
        return `Before or on ${max}`;
```